modules/transport/PostMessageTransport.js

```diff
@@ -0,0 +1,142 @@
+import Postis from 'postis';
+
+/**
+ * The list of methods of incomming postis messages that we have to support for
+ * backward compatability for the users that are directly sending messages to
+ * Jitsi Meet (without using external_api.js)
+ *
+ * @type {string[]}
+ */
+const oldIncommingMethods = [ 'display-name', 'toggle-audio', 'toggle-video',
+    'toggle-film-strip', 'toggle-chat', 'toggle-contact-list',
+    'toggle-share-screen', 'video-hangup', 'email', 'avatar-url' ];
+
+/**
+ * The list of methods of outgoing postis messages that we have to support for
+ * backward compatability for the users that are directly listening to the
+ * postis messages send by Jitsi Meet(without using external_api.js).
+ *
+ * @type {string[]}
+ */
+const oldOutgoingMethods = [ 'display-name-change', 'incoming-message',
+    'outgoing-message', 'participant-joined', 'participant-left',
+    'video-ready-to-close', 'video-conference-joined',
+    'video-conference-left' ];
+
+/**
+ * The postis method used for all messages.
+ *
+ * @type {string}
+ */
+const POSTIS_METHOD_NAME = 'data';
+
+/**
+ * The default options for postis.
+ *
+ * @type {Object}
+ */
+const defaultPostisOptions = {
+    window: window.opener || window.parent
+};
+
+/**
+ * Implements message transport using the postMessage API.
+ */
+export default class PostMessageTransport {
+    /**
+     * Creates new PostMessageTransport instance.
+     *
+     * @param {Object} options - Optional parameters for configuration of the
+     * transport.
+     */
+    constructor(options = {}) {
+        const postisOptions = Object.assign({}, defaultPostisOptions, options);
+
+        this.postis = Postis(postisOptions);
+
+        // backward compatability
+        oldIncommingMethods.forEach(command =>
+            this.postis.listen(command,
+                data => this._onCommandReceived(command, data)));
+
+        this.postis.listen(POSTIS_METHOD_NAME, data =>
+            this._dataReceivedCallBack(data));
+
+        this._dataReceivedCallBack = () => {
+            // do nothing until real callback is set;
+        };
+    }
+
+    /**
+     * Handles incomming data from the transport.
+     *
+     * @param {string} method - The name of the event or request related to the
+     * data received.
+     * @param {Object} data - The data.
+     * @returns {void}
+     */
+    _onCommandReceived(method, data = {}) {
```

Comment:
I changed the name to _onPostisDataReceived because this transport is not working with request/responses and events. The implementation for them is in Transport.js. Do you think this name is OK?